projects/components/src/layout/layout-change.directive.ts

```diff
@@ -1,19 +1,16 @@
-import { AfterViewInit, Directive, EventEmitter, Output } from '@angular/core';
+import { Directive, EventEmitter, Output } from '@angular/core';
 import { LayoutChangeService, SubscriptionLifecycle } from '@hypertrace/common';
 
 @Directive({
   selector: '[htLayoutChange]',
   providers: [SubscriptionLifecycle, LayoutChangeService]
 })
-export class LayoutChangeDirective implements AfterViewInit {
+export class LayoutChangeDirective {
   @Output('htLayoutChange')
   public readonly changeEmitter: EventEmitter<void> = new EventEmitter();
 
   public constructor(private readonly layoutChange: LayoutChangeService, subscriptionLifecycle: SubscriptionLifecycle) {
     subscriptionLifecycle.add(layoutChange.layout$.subscribe(this.changeEmitter));
-  }
-
-  public ngAfterViewInit(): void {
     this.layoutChange.initialize();
```

Comment:
@aaron-steinfeld  This is the fix for the layout change issue. Since we initialize/register the bounding element after view init, during this time the bounding element already has the new width/height. When layout trigger emits a new dom change, such events get filtered out as it doesn't see any change in dimension.

Reply:
Hmm I think I understand, but if everything is sized correctly after view init, do we even need the layout change? We probably shouldn't do our first measurement in any rendering code until after view init.

Reply:
It's not sized correctly. Here is what happens:
1. Gauge component gets initialized.
2. At this time it gets 200px height
3. htlayoutChange directive gets initialized but the host element is not registered. 
3. On first input change, Gauge component renders the widget with 200px height
4. Titled content components view changes and it reduces the height of gauge from 200 to 170px
5. htlayoutChange now registers host element with layout service
6. Gauge is now registered in layout service with 170px height
7. htLayoutChangeTrigger triggers a layout change. This publish gets filtered out when it compares the current height of gauge with its registered height.
8. Nothings gets emitted.

Reply:
So if we move the first render in gauge afterViewInit would that fix the issue? And would it make sense? I'm thinking yes since it's sizing sensitive, it also prevents the double render.

Reply:
I have seen a delay when i tried this last week. But shouldn't we fix this from the layout directive itself as it is likely to affect other places as well?

Reply:
Can we try it in onInit instead of the constructor? That would have the same affect, right?

Reply:
lemme try

Reply:
Didn't work.